src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -568,23 +569,46 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
       }
 
       def processAccountsPage(
-          taggedAccounts: List[BlockTagged[AccountsIndex]],
+          taggedAccounts: List[BlockTagged[Map[AccountId, Account]]],
           taggedDelegateKeys: List[BlockTagged[DelegateKeys]]
       ): Future[(Option[Int], Option[Int], List[BlockTagged[DelegateKeys]])] = {
-        for {
+        // we fetch active delegates per block so we can filter out current active bakers
+        // at this point in time and put the information about it in the separate row
+        // (there is no operation like bakers deactivation)
+        val accountsWithHistoryFut = for {
+          activeBakers <- tezosNodeOperator.fetchActiveDelegates(taggedAccounts.map(x => (x.blockLevel, x.blockHash)))
           activatedOperations <- fetchActivationOperationsByLevel(taggedAccounts.map(_.blockLevel).distinct)
           activatedAccounts <- db.run(TezosDb.findActivatedAccountIds)
           updatedTaggedAccounts = updateTaggedAccountsWithIsActivated(
             taggedAccounts,
             activatedOperations,
             activatedAccounts.toList
           )
-          res <- db.run(TezosDb.writeAccountsAndCheckpointDelegates(updatedTaggedAccounts, taggedDelegateKeys)).map {
-            case (accountWrites, accountHistoryWrites, delegateCheckpoints) =>
-              (accountWrites, delegateCheckpoints, taggedDelegateKeys)
+          accountsWithHistory <- Future.traverse(updatedTaggedAccounts) { blockTaggedAccounts =>
+            if (blockTaggedAccounts.blockLevel % lorreConf.blockRightsFetching.cycleSize == 1) {
```

Comment:
What is this check practically verifying?

Reply:
I changed it a bit, but it verifies if we want to get info about deactivated bakers, because we don't want to do it every block, only at the start of the cycle